File: Jongryul/02/1213.js
const { reverse } = require("dns");
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const w = fs.readFileSync(filePath).toString().trim().split("");
w.sort();
const l = w.length;
let i = 0;
const rlt = [];
let tmp = "";
let check = 0;
let check2 = 0;
while (i < l) {
  if (i < l - 1 && w[i] === w[i + 1]) {
    rlt.push(w[i]);
    i += 2;
  } else {
    if (!check) {
      check = 1;
      tmp = w[i];
      i++;
    } else {
      console.log("I'm Sorry Hansoo");
      check2 = 1;
      break;
    }
  }
}

if (!check2) {
  const tmplist = [...rlt].reverse();
  rlt.push(tmp);
  rlt.push(...tmplist);
  console.log(rlt.join(""));
}

File: _Season 1/08th_Competition/Jongryul/02/1213.py
import sys
w = sorted(list(sys.stdin.readline().rstrip()))

l = len(w)
tmp = ''

rlt = []
check = False
i = 0

while i < l:
    if i < l-1 and w[i] == w[i+1]:
        rlt.append(w[i])
        i += 2
    else:
        if not check:
            tmp = w[i]
            i += 1
            check = True
        else:
            print("I'm Sorry Hansoo")
            break
else:
    rlt += [tmp] + list(reversed(rlt))
    if list(reversed(rlt)) == rlt:
        print(''.join(rlt))
    else:
        print("I'm Sorry Hansoo")